src/compiler/crystal/interpreter/repl.cr

```diff
@@ -44,6 +42,19 @@ class Crystal::Repl
     end
   end
 
+  record EvalSuccess, value : Value, warnings : WarningCollection
+  record EvalParserError, warnings : WarningCollection
```

Comment:
**question:** Have you considered merging these two types as `EvalResult`? Their shape is similar.

Reply:
Sure. Using a nilable value would be enough